### PR TITLE
chore: Set up custom ESLint and Prettier configs

### LINKED ISSUE #848: [CHORE] - Set up custom ESLint and Prettier configs
### What work needs to be performed?

There's a few steps to do here:

- First, remove these two blocks from the `devDependencies` in the `package.json`.
```json
    "@angular-eslint/builder": "17.1.1",
    "@angular-eslint/eslint-plugin": "17.1.1",
    "@angular-eslint/eslint-plugin-template": "17.1.1",
    "@angular-eslint/schematics": "17.1.1",
    "@angular-eslint/template-parser": "17.1.1",
```
```json
    "@typescript-eslint/eslint-plugin": "6.13.2",
    "@typescript-eslint/parser": "6.13.2",
```

- Second, install the custom configuration:
```bash
pnpm i --save-dev --save-exact @nhcarrigan/eslint-config-angular`
```

- Third, update the `.eslintrc.json` file to be:
```json
{
  "extends": "@nhcarrigan/eslint-config-angular"
}
```

- Fourth, install Prettier and the custom config:
```bash
pnpm i --save-dev --save-exact prettier @nhcarrigan/prettier-config
```

- Fifth, create a `.prettierrc.json` file with:
```json
"@nhcarrigan/prettier-config"
```
> [!NOTE]
> There are no braces in this file. Just that single line of text.

- Sixth, update the `lint` script in the `package.json` file to be:
```json
"lint": "ng lint && prettier src --check",
```

- Seventh, run the `lint` command to confirm that configs are loading correctly (if you see linter errors, that's okay!)
```bash
pnpm run lint
```

> [!WARNING]
> At this point, we recommend creating a commit with these changes and pushing to your repo, just to be safe.

IF THERE ARE LINTER ERRORS:

- First, fix the ones that can be fixed automatically:
```bash
pnpm exec eslint src --fix
```
```bash
pnpm exec prettier src --write
```

- Then, fix the remaining ones manually.

> [!NOTE]
> It is absolutely okay to be stuck on this step. Feel free to create a draft PR and we'll be happy to help you work through it.

### Additional information

This is a good first issue, but it's a complex one. Here's some tips:

- If you are confused about the steps, take a look at this [example commit](https://github.com/nhcarrigan/nhcarrigan.github.io/pull/306/commits/7894e904163f0215ec517378073f446e5d3b18f7)
- As always, feel free to join our [Discord](https://chat.naomi.lgbt) if you get stuck or have questions!

### PR DESCRIPTION
**Pull request by a2937**
# Pull Request

## Description:

I followed all the steps but Eslint gave me some issues with path resolution and I wasn't sure if you wanted the console statement. 

## Related Issue:

Closes #848

### CODE REVIEW

File: src/app/gallery/gallery.component.ts [hunk 1]
```diff
@@ -1,18 +1,28 @@
-import { Component, OnInit } from '@angular/core';
-import { featuredHouses, featuredRooms } from 'src/assets/data/featured';
-import { FeaturedSimsInt } from 'src/assets/interfaces/FeaturedSimsInt';
+import { Component, OnInit } from "@angular/core";
+import { featuredHouses, featuredRooms } from "src/assets/data/featured";
+import { FeaturedSimsInt } from "src/assets/interfaces/FeaturedSimsInt";
```
Comment (naomi-lgbt) [inline src/app/gallery/gallery.component.ts:3]:
```suggestion
import { Component, OnInit } from "@angular/core";

import { featuredHouses, featuredRooms } from "../../assets/data/featured";
import { FeaturedSimsInt } from "../../assets/interfaces/FeaturedSimsInt";
```

Resolves the unresolved path error.

File: src/app/home/home.component.ts [hunk 2]
```diff
@@ -1,15 +1,25 @@
-import { Component, OnInit } from '@angular/core';
-import { introText } from 'src/assets/data/intro';
+import { Component, OnInit } from "@angular/core";
+import { introText } from "src/assets/data/intro";
```
Comment (naomi-lgbt) [inline src/app/home/home.component.ts:2]:
```suggestion
import { Component, OnInit } from "@angular/core";

import { introText } from "../../assets/data/intro";
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
`console.err(err)` can be replaced with `alert(err)`.

File: src/app/app.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
+import { RouterTestingModule } from "@angular/router/testing";
 
-describe('AppComponent', () => {
+import { AppComponent } from "./app.component";
+import { FooterComponent } from "./footer/footer.component";
+import { NavComponent } from "./nav/nav.component";
+
+describe("AppComponent", () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   let component: AppComponent;
```
Comment (naomi-lgbt) [inline src/app/app.component.spec.ts:10]:
```suggestion
```

File: src/app/footer/footer.component.spec.ts [hunk 4]
```diff
@@ -1,42 +1,41 @@
...
+      declarations: [FooterComponent]
     }).compileComponents();
     fixture = TestBed.createComponent(FooterComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
     compiled = fixture.nativeElement;
   });
 
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
   beforeEach(() => {});
```
Comment (naomi-lgbt) [inline src/app/footer/footer.component.spec.ts:21]:
```suggestion
```

File: src/app/footer/footer.component.ts [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
+  selector: "app-footer",
+  templateUrl: "./footer.component.html",
+  styleUrls: ["./footer.component.css"]
 })
 export class FooterComponent implements OnInit {
+  /**
+   *
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
```
Comment (naomi-lgbt) [inline src/app/footer/footer.component.ts:16]:
```suggestion
```

File: src/app/gallery/gallery.component.spec.ts [hunk 6]
```diff
@@ -1,72 +1,73 @@
...
+      declarations: [GalleryComponent]
     }).compileComponents();
     fixture = TestBed.createComponent(GalleryComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
     compiled = fixture.nativeElement;
   });
 
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
   beforeEach(() => {});
```
Comment (naomi-lgbt) [inline src/app/gallery/gallery.component.spec.ts:22]:
```suggestion
```

File: src/app/gallery/gallery.component.ts [hunk 7]
```diff
@@ -1,18 +1,29 @@
...
 })
 export class GalleryComponent implements OnInit {
   public houses: FeaturedSimsInt[] = [];
   public rooms: FeaturedSimsInt[] = [];
 
+  /**
+   *
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
   constructor() {}
```
Comment (naomi-lgbt) [inline src/app/gallery/gallery.component.ts:22]:
```suggestion
```

File: src/app/home/home.component.ts [hunk 8]
```diff
@@ -1,15 +1,26 @@
...
+  templateUrl: "./home.component.html",
+  styleUrls: ["./home.component.css"]
 })
 export class HomeComponent implements OnInit {
   public intro: string[] = [];
+  /**
+   *
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
   constructor() {}
```
Comment (naomi-lgbt) [inline src/app/home/home.component.ts:19]:
```suggestion
```

File: src/app/nav/nav.component.ts [hunk 9]
```diff
@@ -1,16 +1,24 @@
...
+  selector: "app-nav",
+  templateUrl: "./nav.component.html",
+  styleUrls: ["./nav.component.css"]
 })
 export class NavComponent implements OnInit {
+  /**
+   *
+   */
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  constructor() {}
```
Comment (naomi-lgbt) [inline src/app/nav/nav.component.ts:16]:
```suggestion
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
I think these should all be safe to remove. Didn't test locally tho.

(Dang this code is ancient)

File: src/app/app.component.spec.ts (hunk 3 above)
Comment (a2937) [reply src/app/app.component.spec.ts:10]: Actually Naomi you assigned it a value. If you directly remove it like that, it'll break. But the test code seems to be in need of an update, which is out of scope for this PR.

Comment (a2937) [reply src/app/app.component.spec.ts:10]: Ideally I'd like to stay until a new PR gets made to redo the tests.

Comment (naomi-lgbt) [reply src/app/app.component.spec.ts:10]: If the assigned value isn't used either, we can just... remove that too.

Comment (a2937) [reply src/app/app.component.spec.ts:10]: Okay done.

File: src/app/app.component.spec.ts [hunk 10]
```diff
@@ -1,25 +1,25 @@
...
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { RouterTestingModule } from "@angular/router/testing";
 
-describe('AppComponent', () => {
+import { AppComponent } from "./app.component";
+import { FooterComponent } from "./footer/footer.component";
+import { NavComponent } from "./nav/nav.component";
+
+describe("AppComponent", () => {
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (naomi-lgbt) [inline src/app/app.component.spec.ts:9]:
```suggestion
```

File: src/app/app.component.spec.ts [hunk 11]
```diff
@@ -1,25 +1,24 @@
...
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { RouterTestingModule } from "@angular/router/testing";
 
-describe('AppComponent', () => {
+import { AppComponent } from "./app.component";
+import { FooterComponent } from "./footer/footer.component";
+import { NavComponent } from "./nav/nav.component";
+
+describe("AppComponent", () => {
   let component: AppComponent;
```
Comment (naomi-lgbt) [inline src/app/app.component.spec.ts:9]: Oh I see what happened here.

`component` needs to be removed - it's unused.
`compiled` needs to be put back, that one's actually used in the tests,

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Couple of the files need a prettier run again :3

File: src/app/app.component.spec.ts (hunk 11 above)
Comment (a2937) [reply src/app/app.component.spec.ts]: Just fixed.

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**